tests/PHPStan/Analyser/nsrt/array-replace.php

```diff
@@ -23,9 +23,9 @@ public function arrayReplace($array1, $array2): void
 	 */
 	public function arrayReplaceArrayShapes($array1, $array2): void
 	{
-		assertType("non-empty-array<'bar'|'foo', '1'|'2'>", array_replace($array1));
-		assertType("non-empty-array<'bar'|'foo', '1'|'2'>", array_replace([], $array1));
-		assertType("non-empty-array<'bar'|'foo', '1'|'2'|'4'>", array_replace($array1, $array2));
+		assertType("array{foo: '1', bar: '2'}", array_replace($array1));
+		assertType("array{foo: '1', bar: '2'}", array_replace([], $array1));
+		assertType("array{foo: '1', bar: '4'}", array_replace($array1, $array2));
 	}
```

Comment:
Can you add another test cases where `$array2` is a general array, please? And maybe in the other section somewhere also one where `$array1` is a general array but `$array2` is a constant array.

a bit more mixed keys (strings or ints) would be interesting too 😊 that would also help massively with potential future refactoring